sympy/integrals/intpoly.py

```diff
@@ -0,0 +1,487 @@
+                x0[max_gen] = b/a[max_gen]
+                x0[gen2] = 0
+            else:
+                x0[max_gen] = b1
+                x0[gen2] = a1
+    else:
+        x0 = {gens[0]: b/a[gens[0]]}
+
+    return x0
+
+
+def decompose(expr):
+    """Decomposes an input polynomial into homogeneous ones of
+    smaller or equal degree.
+
+    Returns a dictionary with keys as the degree of the smaller
+    constituting polynomials. Values are the constituting polynomials.
+
+    Parameters
+    ==========
+    expr : Polynomial(SymPy expression)
+
+    Examples
+    ========
+    >>> from sympy.abc import x, y
+    >>> from sympy.integrals.intpoly import decompose
+    >>> decompose(x**2 + x*y + x + y + x**3*y**2 + y**5)
+    {1: x + y, 2: x**2 + x*y, 5: x**3*y**2 + y**5}
+
+    """
+    poly_dict = {}
+    if isinstance(expr, Expr):
+        if expr.is_Add:
+            for monomial in expr.args:
+                degree = 0
+                if monomial.is_Pow:
+                    degree += monomial.args[1]
+                else:
+                    term_type = len(monomial.args)
+                    if term_type == 0:
+                        if monomial.is_Symbol:
+                            degree += 1
+                        else:
+                            continue
+                    for univariate in monomial.args:
+                        term_type = len(univariate.args)
+                        if term_type == 0 and univariate.is_Symbol:
+                            degree += 1
+                        elif term_type == 2:
+                            degree += univariate.args[1]
+                if degree in poly_dict:
+                    poly_dict[degree] += monomial
+                else:
+                    poly_dict[degree] = monomial
+        elif expr.is_Mul:
+            degree = 0
+            for term in expr.args:
+                term_type = len(term.args)
+                if term_type == 0 and term.is_Symbol:
+                    degree += 1
+                elif term_type == 2:
+                    degree += term.args[1]
+            poly_dict[degree] = expr
+        elif expr.is_Pow:
+            poly_dict[expr.args[1]] = expr
+        else:
+            poly_dict[1] = expr
+    else:
+        poly_dict = {0: expr}
+    return poly_dict
+
+
+def norm(point):
+    """Returns the Euclidean norm of a point from origin.
+
+    Parameters
+    ==========
+    point: This denotes a point in the dimensional space.
+
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import norm
+    >>> from sympy.geometry.point import Point
+    >>> norm(Point(2, 7))
+    7.28010988928052
+
+    """
+    if isinstance(point, tuple):
+        return (point[0] ** 2 + point[1] ** 2)**(1/2)
+    elif isinstance(point, Point):
+        return (point.x ** 2 + point.y ** 2) ** (1 / 2)
+    elif isinstance(point, dict):
+        s = 0
+        for i in point.values():
+            s += i ** 2
+        return s**(1/2)
```

Comment:
This is a common mistake, see http://docs.sympy.org/dev/gotchas.html#python-numbers-vs-sympy-numbers, you have to use `S(1)/2`.